packages/@angular/cli/commands/add.ts

```diff
@@ -0,0 +1,102 @@
+import chalk from 'chalk';
+import { Command, CommandScope, Option } from '../models/command';
+import { parseOptions } from '../models/command-runner';
+import { CliConfig } from '../models/config';
+import { SchematicAvailableOptions } from '../tasks/schematic-get-options';
+
+const SilentError = require('silent-error');
+
+
+export default class AddCommand extends Command {
+  readonly name = 'add';
+  readonly description = 'Add support for a library to your project.';
+  scope = CommandScope.inProject;
+  arguments = ['collection'];
+  options: Option[] = [];
+
+  private _rawArgs: any;
+
+  async initializeRaw(args: any): Promise<any> {
+    this._rawArgs = args.slice();
+    return args;
+  }
+
+  public async parseOptions(collectionName: string): Promise<any> {
+    const SchematicGetOptionsTask = require('../tasks/schematic-get-options').default;
+
+    const getOptionsTask = new SchematicGetOptionsTask({
+      ui: this.ui,
+      project: this.project
+    });
+
+    const availableOptions: SchematicAvailableOptions[] = await getOptionsTask.run({
+      schematicName: 'ng-add',
+      collectionName,
+    });
+    let anonymousOptions: string[] = [];
+
+    if (availableOptions) {
+      const nameOption = availableOptions.filter(opt => opt.name === 'name')[0];
```

Comment:
Do we want to allow for `name` to be specially handled for `add` or would it be best left as another option? This name logic exists for the common name scenario usage of `generate`

`ng add some-lib --name foo`

Reply:
We should get rid of the `name` logic for the `generate`. I removed it from add.